### DISCUSSION TITLE
Does OpenTelemetry experimental.instrumentationHook work outside of Vercel hosting?

### DISCUSSION THREAD

**Discussion by littlesandra88** [vercel/next.js#53376] [Help]
### Summary

Dear all =)

I would like to make an OpenTelemetry PoC with the app router. When I read [this doc](https://nextjs.org/docs/app/building-your-application/optimizing/open-telemetry) it says I can enable `experimental.instrumentationHook` and NextJS will wrap all my code, and give me some default traces.

If that is correct understood, then what confuses me is this sentence

> Good to know: We currently support OpenTelemetry bindings only in serverless functions. We don't provide any for edge or client side code.

**Question**

Does that mean that if I use `experimental.instrumentationHook` then it have to be hosted on Vercel?

So if I want to use on the big could providers, then I have to configure OpenTelemetry the manual way, as described [here](https://nextjs.org/docs/app/building-your-application/optimizing/open-telemetry#manual-opentelemetry-configuration)?

Hugs,
Sandra =)

**Comment by chris-erickson** [comment]
Did you ever find an answer? I'm seeing different behavior when an app is running locally with the [directions for the native Otel](https://nextjs.org/docs/app/building-your-application/optimizing/open-telemetry#using-vercelotel) (it doesn't work consistently at all) but it does when running locally.

Putting our prior init code from the pages router inside here causes all kinds of mayhem - on builds it seems like none of the Otel stuff is initialized yet. Wish there were examples of instrumenting and adding custom spans and attributes to all the parts of NextJS!
